cmd/src/sbom_utils.go

```diff
@@ -0,0 +1,209 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// Manifest represents a simplified structure of the Docker manifest response
+type dockerHubManifest struct {
+	Config struct {
+		Digest string `json:"digest"`
+	} `json:"config"`
+}
+
+// TokenResponse represents the JSON response from dockerHub's token service
+type dockerHubTokenResponse struct {
+	Token string `json:"token"`
+}
+
+// getImageDigest returns the sha256 hash for the given image and tag
+// It supports multiple registries
+func getImageDigest(image string, tag string) (string, error) {
+	if strings.HasPrefix(image, "sourcegraph/") {
+		return getImageDigestDockerHub(image, tag)
+	} else if strings.HasPrefix(image, "us-central1-docker.pkg.dev/") {
+		return getImageDigestGcloud(image, tag)
+	} else {
+		return "", fmt.Errorf("unsupported image registry: %s", image)
+	}
+}
+
+//
+// Implement functionality for Docker Hub
+
+// getImageDigestDockerHub returns the sha256 digest for the given image and tag from DockerHub
+func getImageDigestDockerHub(image string, tag string) (string, error) {
+	// Construct the DockerHub manifest URL
+	url := fmt.Sprintf("https://registry-1.docker.io/v2/%s/manifests/%s", image, tag)
+
+	token, err := getDockerHubAuthToken(image)
+	if err != nil {
+		return "", err
+	}
+
+	// Create a new HTTP request with the authorization header
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", err
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+	req.Header.Add("Accept", "Accept: application/vnd.docker.distribution.manifest.v2+json, application/vnd.oci.image.manifest.v1+json")
+
+	// Make the HTTP request
+	client := &http.Client{}
+	resp, err := client.Do(req)
```

Comment:
General convention is to just use `http.DefaultClient`, which can be configured globally with things like a meaningful user agent